File: app/src/main/java/pl/szczodrzynski/edziennik/datamodels/ConverterJsonObject.java
package pl.szczodrzynski.edziennik.datamodels;

import androidx.room.TypeConverter;

import com.google.gson.JsonObject;
import com.google.gson.JsonParser;

public class ConverterJsonObject {

    @TypeConverter
    public static JsonObject toJsonObject(String value) {
        return value == null ? null : new JsonParser().parse(value).getAsJsonObject();
    }

    @TypeConverter
    public static String toString(JsonObject value) {
        return value == null ? null : value.toString();
    }
}

File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/converter/ConverterJsonObject.kt
package pl.szczodrzynski.edziennik.data.db.converter

import androidx.room.TypeConverter
import com.google.gson.JsonObject
import com.google.gson.JsonParser

class ConverterJsonObject {
    @TypeConverter
    fun toJsonObject(value: String?): JsonObject? = value?.let { JsonParser().parse(it).asJsonObject }

    @TypeConverter
    fun toString(value: JsonObject?): String? = value?.toString()
}
